cmd/zfs/zfs_main.c

```diff
@@ -4357,10 +4380,40 @@ zfs_do_send(int argc, char **argv)
 			}
 		}
 
-		zhp = zfs_open(g_zfs, argv[0], ZFS_TYPE_DATASET);
+		/*
```

Comment:
What if we implemented this logic and the logic to find the incremental source of the partial send in libzfs? That way other programs could take advantage of it, and we wouldn't have to do this logic here, in the cmd code. We could add a libzfs_partial_send function that takes argv[0] and figures out everything else, and then invokes the send api directly.

Reply:
Hmm. I see your point here, but I'm a bit conflicted because I also want the interface to libzfs to be as consistent as possible. `zfs_send_one()` takes a zhp so having a separate interface that takes `argv[0]` seems strange. I think in an ideal world, zfs_main.c would not be opening zhp's at all, but I'm not sure this is the place to make that change. Thoughts @pcd1193182 ?

Reply:
I see your point, but I think the value of making this partial send logic accessible to programs not using the command line is valuable enough to be worth being a little different here. That's just my opinion, though, but I understand if others feel strongly the other way.